lib/Remotery.c

```diff
@@ -4233,11 +4233,11 @@ static void Remotery_Destructor(Remotery* rmt)
     // Join the remotery thread before clearing the global object as the thread is profiling itself
     Delete(Thread, rmt->thread);
 
-    // Ensure this is the module that created it
-    assert(g_RemoteryCreated == RMT_TRUE);
-    assert(g_Remotery == rmt);
-    g_Remotery = NULL;
-    g_RemoteryCreated = RMT_FALSE;
+    if (g_RemoteryCreated)
+    {
+      g_Remotery = NULL;
+      g_RemoteryCreated = RMT_FALSE;
+    }
 
```

Comment:
I'm looking at the code and it doesn't look like the `if (g_RemoteryCreated)` check is needed:

- If `g_RemoteryCreated == RMT_FALSE` then there is no harm in setting it to `RMT_FALSE` anyway.
- When `g_RemoteryCreated == RM_FALSE` and `g_Remotery` is a valid pointer (from `rmt_SetGlobalInstance`), the destructor will never be called as the assert below would trap it before then.

Does that make sense?